lib/flipper/adapters/memcached.rb

```diff
@@ -0,0 +1,74 @@
+require 'dalli'
+
+module Flipper
+  module Adapters
+    class Memcached
+      include Flipper::Adapter
+      FeaturesKey =  :flipper_features
+
+      attr_accessor :cache
+      attr_accessor :name
+
+      def initialize(adapter, address = 'localhost:11211', options = {})
+        @adapter = adapter
+        @name = adapter.name
+        @cache = Dalli::Client.new(address, options)
+      end
+
+      def features
+        cached(FeaturesKey) do
+          @adapter.features
+        end
+      end
+
+      def add(feature)
+        @adapter.add(feature).tap do
```

Comment:
Probably just personal preference, but I'm not a fan of tap really. I think my thought is there is no way a block/yield is faster than a temporary variable like this:

```ruby
result = @adapter.add(feature)
@cache.delete(FeaturesKey)
result
```

I also think that it reads more explicitly for any programmer in any language as that is a common idiom, whereas tap might be known just by rubyists and what not. Not a huge concern there, but if things are more common, I don't mind if they are a bit less aesthetically beautiful. 

Any passionate thoughts or feelings on keeping tap?